src/shared/bus-unit-util.c

```diff
@@ -390,6 +494,157 @@ int bus_append_unit_property_assignment(sd_bus_message *m, const char *assignmen
 
                 r = sd_bus_message_append(m, "v", "i", (int32_t) n);
 
+#ifdef HAVE_SECCOMP
+
+        } else if (streq(field, "SystemCallFilter")) {
+                bool whitelist;
+                const char *p;
+
+                r = sd_bus_message_open_container(m, 'v', "bas");
+                if (r < 0)
+                        return bus_log_create_error(r);
+
+                p = eq;
+                if (*p == '~') {
+                        whitelist = false;
+                        p++;
+                } else
+                        whitelist = true;
+
+                r = sd_bus_message_append_basic(m, 'b', &whitelist);
+                if (r < 0)
+                        return bus_log_create_error(r);
+
+                r = sd_bus_message_open_container(m, 'a', "s");
+                if (r < 0)
+                        return bus_log_create_error(r);
+
+                if (whitelist) {
+                        r = sd_bus_message_append_basic(m, 's', "@default");
+                        if (r < 0)
+                                return bus_log_create_error(r);
+                }
+
+                for (;;) {
+                        _cleanup_free_ char *word = NULL;
+                        int af;
+
+                        r = extract_first_word(&p, &word, NULL, EXTRACT_QUOTES);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s value: %s", field, eq);
+                        if (r == 0)
+                                break;
+
+                        af = af_from_name(word);
```

Comment:
you don't mean af_from_name here, really, but syscall_from_name()... that said, given that the local set of syscalls might be quite different from what the server side knows i wouldn't do this validation here, and just pass the strings on as-is, the server side can validate this